include/pulsar/c/consumer_configuration.h

```diff
@@ -89,12 +89,14 @@ typedef enum
     pulsar_consumer_regex_sub_mode_AllTopics = 2
 } pulsar_consumer_regex_subscription_mode;
 
+// Though any field could be non-positive, if all of them are non-positive, this policy will be treated as
+// invalid
 typedef struct {
-    // Max num message, if less than 0, it means no limit.
-    int maxNumMessage;
-    // Max num bytes, if less than 0, it means no limit.
+    // Max num messages, a non-positive value means no limit.
+    int maxNumMessages;
```

Comment:
Refer to your comment on #237 

> In addition, I think we'd better not to use raw struct. In C, the initial field values of a struct is not determined. 

If users do not set this field, then this field is an undefined value.

For examples:
```
pulsar_consumer_batch_receive_policy_t batch_receive_policy{.maxNumeBytes = 1000}
pulsar_consumer_configuration_set_batch_receive_policy(consumer_conf, &batch_receive_policy);
```
The `maxNumMessages` and `timeoutMs` is an indeterminate value.

I think about it. For C users, he should know this behavior about undefined values in the struct.

So, We don't provide a default value to users, and we just to declaration is clear what the behavior is when the value is `xxx`.

Do you agree with me?

Reply:
Yes, see my additional explanations here:
- https://github.com/apache/pulsar-client-cpp/pull/237#discussion_r1172743122
- https://github.com/apache/pulsar-client-cpp/pull/237#discussion_r1172745690